dango/lending/src/query.rs

```diff
@@ -64,3 +77,64 @@ pub fn query_debts(
         .take(limit as usize)
         .collect()
 }
+
+pub fn query_collateral_powers(
+    storage: &dyn Storage,
+) -> StdResult<BTreeMap<Denom, CollateralPower>> {
+    COLLATERAL_POWERS.load(storage)
+}
+
+// Mock oracle price for now.
+// TODO: Update once we finish oracle implementation.
+// Perhaps we want to make a "oracle querier" that has a cache? So that we don't
+// have to query the price for the same denom twice.
+pub fn query_oracle_price(_querier: &QuerierWrapper, _denom: &Denom) -> StdResult<Udec128> {
+    Ok(Udec128::ONE)
+}
+
+/// Calculates the health of a margin account.
+pub fn calculate_account_health(
+    querier: &QuerierWrapper,
+    margin_account: Addr,
+    debts: Coins,
+    collateral_powers: BTreeMap<Denom, CollateralPower>,
+) -> anyhow::Result<HealthResponse> {
+    // Calculate the total value of the debts.
+    let mut total_debt_value = Udec128::ZERO;
+    for coin in debts {
+        let price = query_oracle_price(querier, &coin.denom)?;
+        total_debt_value += coin.amount.checked_into_dec()? * price;
```

Comment:
use dedicated function to mul `Int` with `Dec`
```suggestion
        total_debt_value += coin.amount.checked_mul_dec(price)?;
```

Reply:
But that gives an Int? Is there an equivalent that gives Dec?